mypy/parse.py

```diff
@@ -821,7 +839,12 @@ def construct_function_type(self, args: List[Argument], ret_type: Type,
         if ret_type is None:
             ret_type = AnyType(implicit=True)
         arg_kinds = [arg.kind for arg in args]
-        arg_names = [arg.variable.name() for arg in args]
+        if include_names:
+            arg_names = [arg.variable.name()
```

Comment:
This implements "names starting with __ are unnamed arguments".  It's probably necessary to actually get the typeshed files passing.